src/scales/quantitativeScale.ts

```diff
@@ -20,10 +21,89 @@ module Plottable {
      */
     constructor(scale: D3.Scale.QuantitativeScale) {
       super(scale);
+      this._paddingExceptions = new Utils.Map<any, D>();
+      this._includedValues = new Utils.Map<any, D>();
     }
 
     protected _getExtent(): D[] {
-      return this._domainer.computeDomain(this._getAllExtents(), this);
+      var extents = this._getAllExtents().filter((extent) => extent.length > 0);
+      var defaultExtent = this._defaultExtent();
+      if (extents.length === 0) {
+        return defaultExtent;
+      }
+      var combinedExtent = [
+        Utils.Methods.min<D[], D>(extents, (extent) => extent[0], defaultExtent[0]),
+        Utils.Methods.max<D[], D>(extents, (extent) => extent[1], defaultExtent[1])
+      ];
+      var includedDomain = this._includeValues(combinedExtent);
+      var paddedDomain  = this._padDomain(includedDomain);
+      return paddedDomain;
+    }
+
+    public addPaddingException(key: any, exception: D) {
+      this._paddingExceptions.set(key, exception);
+      this._autoDomainIfAutomaticMode();
+      return this;
+    }
+
+    public removePaddingException(key: any) {
+      this._paddingExceptions.delete(key);
+      this._autoDomainIfAutomaticMode();
+      return this;
+    }
+
+    public addIncludedValue(key: any, value: D) {
+      this._includedValues.set(key, value);
+      this._autoDomainIfAutomaticMode();
+      return this;
+    }
+
+    public removeIncludedValue(key: any) {
+      this._includedValues.delete(key);
+      this._autoDomainIfAutomaticMode();
+      return this;
+    }
+
+    public padProportion(): number;
+    public padProportion(padProportion: number): QuantitativeScale<D>;
+    public padProportion(padProportion?: number): any {
+      if (padProportion == null) {
+        return this._padProportion;
+      }
+      if (padProportion < 0) {
+        throw Error("padProportion must be non-negative");
```

Comment:
Actually, it looks like invoking it with and without `new` do the same thing: http://es5.github.io/#x15.11.1

I think we should stick with one style though, so I'll add the `new`.